WordPress/gutenberg
# Refine rename flow for blocks with overrides

## What?

Part of https://github.com/WordPress/gutenberg/issues/59813. Supersedes https://github.com/WordPress/gutenberg/pull/60066. Also close https://github.com/WordPress/gutenberg/issues/60118.

**The latest design is https://github.com/WordPress/gutenberg/pull/60234#issuecomment-2034513896**

This PR does a few things:
1. If no name exists on block, and overrides is disabled, use "allow overrides" button
2. If name exists and overrides is disabled, use toggle
3. If overrides is enabled, use toggle
4. When clicking button, show "allow overrides" modal that is essentially a rebranded "rename block" modal
5. In all other rename actions, use the standard rename modal with overrides warning text
6. The default state for overrides is disabled, users have to opt-in to enable pattern overrides

## Why?

This is to simplify the UI flow for renaming and toggling overrides for blocks.

## How?


Add overrides warning to the rename modal and change the overrides control to a button when there's no name.

## Testing Instructions




1. Create a synced pattern and add a few paragraphs.
2. If the block has no name, the control should show a button for overrides.
3. Clicking on the button should open a renaming modal.
4. Once the block is named, the control becomes a toggle button.
5. Other renaming flow (list view, block toolbar, shortcut) should show overrides warning.

## Screenshots or screencast 

https://github.com/WordPress/gutenberg/assets/7753001/37ed9f09-0b7b-447d-835f-00dcabc98077
